sunpy/net/jsoc/jsoc.py

```diff
@@ -40,10 +40,24 @@ def __init__(self, table=None):
         """
         table : `astropy.table.Table`
         """
-
-        self.table = table
-        self.query_args = None
-        self.requests = None
+        if isinstance(table, list):
+            # fido_factory._handle_record_slice
+            # will return a list containing the
+            # response that has been indexed,
+            # so we have to remove that.
+            new_table = table[0]
+            if isinstance(new_table, type(self)):
+                self.table = new_table.table
+                self.query_args = new_table.query_args
+                self.requests = new_table.requests
+            else:
+                self.table = table
+                self.query_args = None
+                self.requests = None
+        else:
+            self.table = table
+            self.query_args = None
+            self.requests = None
```

Comment:
is there a different way to repeat all these lines multiple times?
What about define them at first, as such, and then only change them when needed?